### PR TITLE
Add swift-testing

### PR DESCRIPTION
**Pull request by justice-adams-apple**
### Pull Request Description
* Add swift-testing to the source compat suite

### CODE REVIEW

**Comment by justice-adams-apple** [comment]
@swift-ci test

File: projects.json [hunk 1]
```diff
@@ -4620,5 +4620,39 @@
         "action": "TestSwiftPackage"
       }
     ]
+  },
+  {
+    "repository": "Git",
+    "url": "https://github.com/apple/swift-testing",
```
Comment (justice-adams-apple) [inline projects.json:4626]: cc @ahoppen will adding this project cause issues for the stress tester?

Comment (ahoppen) [reply projects.json:4626]: Since you don’t put the `source kit-smoke` tag on there, I think it should be fine. Thanks for checking.

**Comment by justice-adams-apple** [comment]
@swift-ci test

File: projects.json [hunk 2]
```diff
@@ -4620,5 +4620,39 @@
...
+        "action": "BuildSwiftPackage",
+        "configuration": "release",
+        "build_tests": "true",
+        "build_tests_release": "true",
+        "tags": "swiftpm",
+        "xfail": [
+          {
+            "issue": "https://github.com/apple/swift/issues/69096",
+            "compatibility": ["5.0"],
+            "branch": ["release/5.10", "main"],
```
Comment (stmontgomery) [inline projects.json:4651]: Note that our current expected failure is with `main` toolchains only, not 5.10 toolchains. Should this be adjusted to the following? (Apologies if I'm misinterpreting what this means)
```suggestion
            "branch": ["main"],
```

Comment (justice-adams-apple) [reply projects.json:4651]: Will update 👍

**Comment by justice-adams-apple** [comment]
@swift-ci test

**Comment by justice-adams-apple** [comment]
cc @shahmishal this should be ready to land

**Review by shahmishal — APPROVED** [review]

**Pull request merged**